api.R

```diff
@@ -11,91 +12,130 @@ source("generics.R")
 # Read AWS creds from Docker secrets
 if (file.exists("/run/secrets/ENV_FILE")) {
   readRenviron("/run/secrets/ENV_FILE")
-} else {
+} else if (file.exists("secrets/ENV_FILE")) {
   readRenviron("secrets/ENV_FILE")
+} else {
+  readRenviron(".env")
 }
-readRenviron(".env")
 
 # Get the model run attributes at runtime from env vars
 dvc_bucket <- Sys.getenv("AWS_S3_DVC_BUCKET")
 run_bucket <- Sys.getenv("AWS_S3_MODEL_BUCKET")
-run_id <- Sys.getenv("AWS_S3_MODEL_RUN_ID")
-run_year <- Sys.getenv("AWS_S3_MODEL_YEAR")
 api_port <- as.numeric(Sys.getenv("API_PORT", unset = "3636"))
+default_run_id_var_name <- "AWS_S3_MODEL_RUN_ID"
+default_run_id <- Sys.getenv(default_run_id_var_name)
 
-
-# Download Files ---------------------------------------------------------------
-
-# Grab model fit and recipe objects
-temp_file_fit <- tempfile(fileext = ".zip")
-aws.s3::save_object(
-  object = file.path(
-    run_bucket, "workflow/fit",
-    paste0("year=", run_year),
-    paste0(run_id, ".zip")
-  ),
-  file = temp_file_fit
+# The list of run IDs that will be deployed as possible model endpoints
+valid_run_ids <- c(
+  "2024-02-06-relaxed-tristan",
+  "2024-03-17-stupefied-maya"
```

Comment:
Any other run IDs that we should include in this vector for now?

Reply:
Cool, done in abe540391c05b750507e5086fedbf27b31c49d51. Are the old workbooks even still operational, given that the model version in the (currently static) API has changed since 2022 and 2023?